lib/puppet/provider/ssl_pkey/openssl.rb

```diff
@@ -10,17 +10,17 @@ def self.dirname(resource)
   end
 
   def self.generate_key(resource)
+    options = {}
     case resource[:authentication]
     when :dsa
-      OpenSSL::PKey::DSA.new(resource[:size])
+      options[:dsa_paramgen_bits] = resource[:size] if resource[:size]
```

Comment:
Isn't it time to drop DSA? We need to do a major release anyway and there's a reason why modern distributions don't support it anymore or block it.

Reply:
Fair point. https://github.com/voxpupuli/puppet-openssl/pull/222